packages/file-storage-browser/src/index.ts

```diff
@@ -206,3 +211,33 @@ async function getDirectDownloadInfo(
   const result = await api.directAttachmentsIdDownloadUrlGet(id, type);
   return result.attachment_direct_download_url;
 }
+
+export function fileUploadBrowser({
+  file,
+  vaultUrl,
+  authConfig,
+  videoCodec,
+  progressUpdateFunc = null,
+}: {
+  file: File;
+  vaultUrl: string;
+  authConfig: IFileStorageAuthConfiguration;
+  videoCodec?: string;
+  progressUpdateFunc?:
+    | ((chunkBuffer: ArrayBuffer | null, percentageComplete: number) => void)
+    | null;
+}) {
+  let cancel;
+  const promise = new Promise((resolve, reject) => (cancel = resolve));
```

Comment:
I think this should be a lambda that has the cancel string baked into it so the end user doesn't have to call it as `cancel('cancel')`

```suggestion
  const promise = new Promise((resolve, reject) => (cancel = () => resolve('cancel')));
```